martin/src/config/file/main.rs

```diff
@@ -66,11 +66,15 @@ pub struct Config {
     ///
     /// Can be overridden by [`tile_cache_size_mb`](Self::tile_cache_size_mb) or similar configuration options.
     pub cache_size_mb: Option<u64>,
+
     /// Maximum size of the tile cache in megabytes (0 to disable)
     ///
     /// Overrides [`cache_size_mb`](Self::cache_size_mb)
     pub tile_cache_size_mb: Option<u64>,
 
+    #[serde(default)]
+    pub fail_on_source_resolution_warnings: bool,
```

Comment:
I kind of would like this to be an enum as following, as this way this is not a super specific option, but rather a more genererl setting.

This way we are a bit more explicit what happens.

```suggestion
    #[serde(default)]
    pub on_invalid: OnInvalid,
```

```rust
#[derive(PartialEq, Eq, Debug, Clone, Copy, Default, Serialize, Deserialize, ValueEnum)]
#[serde(rename_all = "lowercase")]
pub enum OnInvalid {
    /// Print warning message, and abort if the error is critical
    #[default]
    Warn,
    /// Do not start Martin on any warnings
    Abort,
    /// Ignore even critical errors
    Ignore,
}
```